src/tool_operate.c

```diff
@@ -506,11 +507,22 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
         retry = RETRY_FTP;
     }
 
+    if(result && !retry && config->retry_all_errors) {
+      /* Warn the first time --retry-all-errors is used for the transfer. */
+      if(!per->config->retry_all_errors_warned) {
+        warnf(global,
+              "Option --retry-all-errors may have unintended consequences.\n");
```

Comment:
I'm not sure I think outputting a warning here just for the option being used is a good idea.